src/executor/mod.rs

```diff
@@ -49,7 +49,120 @@
+    //!     }));
+    //!
+    //!     Ok::<_, ()>(())
+    //! }));
+    //! # }
+    //! ```
+    //!
+    //! The `block_on_all` function will block the current thread until **all**
+    //! tasks that have been spawned onto the [`CurrentThread`] instance have
+    //! completed.
+    //!
+    //! More fine-grain control can be achieved by using [`CurrentThread`] directly.
+    //!
+    //! ```
+    //! # extern crate tokio;
+    //! # extern crate futures;
+    //! # use tokio::executor::current_thread::CurrentThread;
+    //! use futures::future::{lazy, empty};
+    //! use std::time::Duration;
+    //!
+    //! // Calling execute here results in a panic
+    //! // current_thread::spawn(my_future);
+    //!
+    //! # pub fn main() {
+    //! let mut current_thread = CurrentThread::new();
+    //!
+    //! // Spawn a task, the task is not executed yet.
+    //! current_thread.spawn(lazy(|| {
+    //!     println!("Spawning a task");
+    //!     Ok(())
+    //! }));
+    //!
+    //! // Spawn a task that never completes
+    //! current_thread.spawn(empty());
+    //!
+    //! // Run the executor, but only until the provided future completes. This
+    //! // provides the opportunity to start executing previously spawned tasks.
+    //! let res = current_thread.block_on(lazy(|| {
+    //!     Ok::<_, ()>("Hello")
+    //! })).unwrap();
+    //!
+    //! // Now, run the executor for *at most* 1 second. Since a task was spawned
+    //! // that never completes, this function will return with an error.
+    //! current_thread.run_timeout(Duration::from_secs(1)).unwrap_err();
+    //! # }
+    //! ```
+    //!
+    //! # Execution model
+    //!
+    //! Internally, [`CurrentThread`] maintains a queue. When one of its tasks is
+    //! notified, the task gets added to the queue. The executor will pop tasks from
+    //! the queue and call [`Future::poll`]. If the task gets notified while it is
+    //! being executed, it won't get re-executed until all other tasks currently in
+    //! the queue get polled.
+    //!
+    //! Before the task is polled, a thread-local variable referencing the current
+    //! [`CurrentThread`] instance is set. This enables [`spawn`] to spawn new tasks
+    //! onto the same executor without having to thread through a handle value.
+    //!
+    //! If the [`CurrentThread`] instance still has uncompleted tasks, but none of
+    //! these tasks are ready to be polled, the current thread is put to sleep. When
+    //! a task is notified, the thread is woken up and processing resumes.
+    //!
+    //! All tasks managed by [`CurrentThread`] remain on the current thread. When a
+    //! task completes, it is dropped.
+    //!
+    //! [`spawn`]: fn.spawn.html
+    //! [`block_on_all`]: fn.block_on_all.html
+    //! [`CurrentThread`]: struct.CurrentThread.html
+    //! [`Future::poll`]: https://docs.rs/futures/0.1/futures/future/trait.Future.html#tymethod.poll
+
+    pub use tokio_current_thread::{
```

Comment:
It looks like there are some re-exports missing (https://docs.rs/tokio/0.1.6/tokio/executor/current_thread/index.html)

* `BlockError`
* `RunTimeoutError`
* `Turn`
* `TurnError`
* `spawn`

Reply:
Ok, I've added re-exports for all missing public struct and fn which are not deprecated.

Reply:
Actually, that is a good point.

The deprecated APIs have to stay in `tokio::executor::current_thread`, but they don't actually have to move to the new crate. This is to maintain backwards compatibility.

Is it possible to do that? (keep them in the old location w/o introducing them in thew new ones).

Reply:
You're right it must stay in the old location to no break code which still use the deprecated API. I will take a look on it at the end of the week, and will then update the PR.

Thanks for the review !